vagrant/bin/vhosts.sh

```diff
@@ -59,6 +80,49 @@ for site in $(find $sitesdir -maxdepth 1 -type d); do
         fi
     done
 done
+
+# ngnix ssl detection loop
+for site in $(find $sitesdir -maxdepth 1 -type d); do
+    hostname="$(basename $site)"
+    sslconffile="$sslconfdir/$hostname.conf"
+
+    # custom ssl config
+    if [[ -f "$site/$sslconfcust" ]]; then
+        # if the file exists and is identical, don't bother replacing it
+        if [[ -f "$sslconffile" ]] && cmp "$sslconffile" "$site/$sslconfcust" > /dev/null; then
+            continue
+        fi
+
+        if [[ -f "$sslconffile" ]]; then
+            echo "    configured $hostname for ssl (custom vhost was updated)"
+        else
+            echo "    configured $hostname for ssl (custom vhost)"
+        fi
+
+        cp "$site/$sslconfcust" "$sslconffile"
+        continue
+    fi
+
+    for try in $(echo "pub html htdocs"); do
+        pubdir="${site}/${try}"
+        if [[ -d "$pubdir" ]]; then
+            pubname=$(basename $pubdir)
+
+            if [[ -f "$sslconffile" ]]; then
+                break
+            fi
+
+            cp "$ssltemplate" "$sslconffile"
+            perl -pi -e "s/__HOSTNAME__/$hostname/g" "$sslconffile"
+            perl -pi -e "s/__PUBNAME__/$pubname/g" "$sslconffile"
+
+            generate_ssl_cert $hostname 2> /dev/null
```

Comment:
$hostname should be quoted. Since it's a value coming from the file-system, it's a pre-caution to avoid argument poisoning. Minimal risk since hostnames _shouldn't_ have spaces in them, but… ya never know ;)

    generate_ssl_cert "$hostname" 2> /dev/null